### PR TITLE
Add message validations

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

To validate the window post messages to avoid parsing messages that the code doesn't know what to do with them.

# Changes

* Add validations in the handle function.
* Add comments to explain where the validation comes from.

# Tests

* Added a test with "malicious" window messages from another origin and check that those don't affect the regular flow.

### CODE REVIEW

**Comment by lmuntaner** [comment]
* **#11** <a href="https://app.graphite.dev/github/pr/dfinity/verifiable-credentials-sdk/11?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lmuntaner and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js-library/src/request-verifiable-presentation.ts [hunk 1]
```diff
@@ -37,13 +38,13 @@ export type CredentialsRequest = {
 /**
  * Helper functions
  */
-// TODO: Support multiple flows at the same time.
```
Comment (lmuntaner) [inline js-library/src/request-verifiable-presentation.ts:40]: I forgot to remove it in the last PR.

File: js-library/src/request-verifiable-presentation.ts [hunk 2]
```diff
@@ -100,11 +124,24 @@ export const requestVerifiablePresentation = ({
...
   const handleFlowFactory = (currentFlowId: FlowId) => (evnt: MessageEvent) => {
-    // Check how AuthClient does it: https://github.com/dfinity/agent-js/blob/a51bd5b837fd5f98daca5a45dfc4a060a315e62e/packages/auth-client/src/index.ts#L504
-    if (
-      evnt.data?.method === "vc-flow-ready" &&
-      !currentFlows.has(currentFlowId)
-    ) {
+    // The handler is listening to all window messages.
+    // For example, a browser extension could send messages that we want to ignore.
+    if (evnt.origin !== identityProvider) {
+      console.warn(
```
Comment (frederikrothenberger) [inline js-library/src/request-verifiable-presentation.ts:130]: I would lower this to `info` and reword it a bit (i.e "ignoring message from ..."). This is not something to be alarmed by.

**Review by frederikrothenberger — APPROVED** [review]
LGTM, one minor comment.

File: js-library/src/request-verifiable-presentation.ts (hunk 2 above)
Comment (lmuntaner) [reply js-library/src/request-verifiable-presentation.ts:130]: I used the same as AuthClient: https://github.com/dfinity/agent-js/blob/a51bd5b837fd5f98daca5a45dfc4a060a315e62e/packages/auth-client/src/index.ts#L504

I believe it makes sense that both behave similar.

**Pull request merged**